src/latch_cli/services/launch/launch_v2.py

```diff
@@ -0,0 +1,165 @@
+import base64
+import json
+from json.decoder import JSONDecodeError
+from typing import Any, Optional, Union, get_args, get_origin
+
+import dill
+import google.protobuf.json_format as gpjson
+from flyteidl.core import interface_pb2 as _interface_pb2
+from flytekit.core.context_manager import FlyteContextManager
+from flytekit.core.promise import translate_inputs_to_literals
+from flytekit.models.interface import Variable, VariableMap
+
+from latch.utils import current_workspace
+from latch_cli import tinyrequests
+from latch_cli.services.launch.interface import get_workflow_interface
+from latch_cli.utils import get_auth_header
+from latch_sdk_config.latch import config
+
+
+def launch(*, wf_name: str, params: dict[str, Any], version: Optional[str] = None) -> int:
+    """Create an execution of workflow `wf_name` with version `version` and parameters `params`.
+
+    If version is not provided, the latest version of the workflow will be launched.
+
+    This command is not backwards compatible with workflows registered with latch version < 2.62.0 in the container.
+
+    Args:
+        wf_name: Name of workflow to launch (see `.latch/workflow_name` in the workflow directory).
+        params: A dictionary of parameters to pass to the workflow.
+        version: An optional workflow version to launch, defaulting to latest.
+
+    Returns:
+        Execution ID of the launched workflow.
+    """
+    target_account_id = current_workspace()
+
+    wf_id, interface, _ = get_workflow_interface(target_account_id, wf_name, version)
+
+    flyte_interface_types: dict[str, Variable] = VariableMap.from_flyte_idl(gpjson.ParseDict(interface, _interface_pb2.VariableMap())).variables
+
+    python_interface_with_defaults: Union[dict[str, tuple[type, Any]], None] = None
+    for v in flyte_interface_types.values():
+        description: dict[str, Any] = json.loads(v.description)
+        if description.get("idx") != 0:
+            continue
+
+        raw_python_interface_with_defaults = description.get("__workflow_meta__", {}).get("meta", {}).get("python_interface")
+        if raw_python_interface_with_defaults is not None:
+            python_interface_with_defaults = dill.loads(base64.b64decode(raw_python_interface_with_defaults))  # noqa: S301
+            break
+
+    if python_interface_with_defaults is None:
+        raise ValueError("No python interface found -- re-register workflow with latch version >= 2.62.0 in workflow Dockerfile")
```

Comment:
"workflow was registered with an outdated version of latch. Re-register with `latch >= 2.62.0` in workflow Dockerfile or use the legacy `latch launch` command"